pyproject_metadata/__init__.py

```diff
@@ -113,46 +115,36 @@ class ConfigurationWarning(UserWarning):
     """Warnings about backend metadata."""
 
 
-class RFC822Message:
-    """Python-flavored RFC 822 message implementation."""
-
-    __slots__ = ('_headers', '_body')
-
-    def __init__(self) -> None:
-        self._headers: list[tuple[str, str]] = []
-        self._body: str | None = None
+@dataclasses.dataclass
+class _SmartMessageSetter:
+    """
+    This provides a nice internal API for setting values in an RFC822Message to
+    reduce boilerplate.
 
-    def items(self) -> list[tuple[str, str]]:
-        return self._headers.copy()
+    If a value is None, do nothing.
+    If a value contains a newline, indent it (may produce a warning in the future).
```

Comment:
Uh? Do we actually need this? Why? Does this even work with `EmailMesage`? I would not be surprised if it normalized whitespace in headers or encode it somehow. Anyhow, how do multi-line strings end up here?

Reply:
It produces the same thing we produce now. It was added in #2, and pretty sure it mostly shows up with multiline summaries:

* https://github.com/pypa/pyproject-metadata/issues/30
* https://github.com/pypa/pyproject-metadata/pull/112

I think we could add a warning for now, and then not include the behavior if we can make backward incompatible changes (#140).

Reply:
(Note that in #2, the extended mechanism is only for the description, which is now in the body instead, and the actual mechanism included pipe chars).

Reply:
(and in scikit-build-core I manually disallow multiline summaries)

Reply:
And you can actually read these multiline strings with `email.parse`, I did check it, it keeps the indention. Without indentation, they don't work, of course.

Reply:
I pulled out and merged the preparation steps, so you can see just the move by itself.